packages/core/test/injector/helpers/provider-classifier.spec.ts

```
@@ -0,0 +1,132 @@
+import { ClassProvider, FactoryProvider, ValueProvider } from '@nestjs/common';
+import { expect } from 'chai';
+import {
+  isClassProvider,
+  isFactoryProvider,
+  isValueProvider,
+} from '../../../injector/helpers/provider-classifier';
+
+describe('provider classifier', () => {
+  describe('isClassProvider', () => {
```

For each of these utility functions, it may be useful to have some runtime checks to guarantee we're dealing with correct providers. For instance, in the case of a `FactoryProvider` that provides something like:
```ts
const wrongFactoryProvider = {
  provide: 'token',
  useFactory: {}
} as FactoryProvider
```  

The object itself is not a real function. Should this really be recognized as a `FactoryProvider` in this case? Or we should rather throw an error informing the given value is not a real function?